src/stactools/sentinel1/grd/product_metadata.py

```diff
@@ -85,11 +93,11 @@ def scene_id(self) -> str:
     def product_id(self) -> str:
         # Parse the name from href as it doesn't exist in xml files
         href = self.href
-        result = href.split("/")[-2]
+        result = str(href.split("/")[-2])
         if result is None:
             raise ValueError(
-                "Cannot determine product ID using product metadata "
-                f"at {self.href}")
+                "Cannot determine product ID using product metadata " f"at {self.href}"
+            )
         else:
```

Comment:
Actually, since the `if` will _never_ be entered (because `result` _cannot_ be `None`), simply eliminate the _entire_ if/else, as well as the assignment to `result`, and simply do this, if you don't care to deal with the possible `IndexError`:

```
return href.split("/")[-2]
```